fvm/src/kernel/default.rs

```diff
@@ -275,11 +275,11 @@ where
         Ok((id, stat))
     }
 
-    fn block_create(&mut self, codec: u64, data: &[u8]) -> Result<BlockId> {
+    fn block_create(&mut self, codec: u64, data: &(impl AsRef<[u8]> + ?Sized)) -> Result<BlockId> {
         self.call_manager
```

Comment:
note: one would usually call `let data = data.as_ref()` at the top.

Reply:
Why:

- Can leave the generics out of the rest of the function.
- Maybe the compiler will get smart and split the function into a monomorphized portion, and a shared portion.